deltachat-server/rpc-client.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+import asyncio
+import json
+
+
+class Deltachat:
+    def __init__(self, process):
+        self.process = process
+        self.event_queue = asyncio.Queue()
+        self.id = 0
+        self.reader_task = asyncio.create_task(self.reader_loop())
+
+        # Map from request ID to asyncio.Event.
+        # When event happens, it is replaced with a response.
+        self.request_events = {}
+
+    async def reader_loop(self):
+        while True:
+            line = await self.process.stdout.readline()
+            response = json.loads(line)
+            if "id" in response:
+                response_id = response["id"]
+                event = self.request_events[response_id]
+                self.request_events[response_id] = response
```

Comment:
I don't particularly like changing what's in the `request_events` map like this.  I'd maybe put an object inside it that encapsulates the event and data.  I imagine it has a method that says `.response(response)` and it stores the response on the object and sets the event.

or some other way of passing the data back through, just not via the same dict in which the event is stored.